packages/firebase_auth/firebase_auth/lib/src/firebase_auth.dart

```diff
@@ -71,6 +71,39 @@ class FirebaseAuth extends FirebasePluginPlatform {
     return null;
   }
 
+  /// Changes this instance to point to an Auth emulator running locally.
+  ///
+  /// Set the [origin] of the local emulator, such as "http://localhost:9099"
+  ///
+  /// Note: Must be called immediately, prior to accessing auth methods.
+  /// Do not use with production credentials as emulator traffic is not encrypted.
+  Future<void> useEmulator(String origin) async {
+    assert(origin.isNotEmpty);
+    String mappedOrigin = origin;
+
+    // Android considers localhost as 10.0.2.2 - automatically handle this for users.
+    if (defaultTargetPlatform == TargetPlatform.android) {
```

Comment:
I don't see this block (and the `match == null`) covered by tests. Should we add tests for those?

Reply:
Mmm - I am unable to enable the e2e testing because the auth e2e test suite needs a conversion to local emulator style in order to execute.

I test the regex here https://github.com/FirebaseExtended/flutterfire/pull/4263/files/4eeec0c8114ff1dbb36f50d83be68abb4b197e09#diff-8f471241042de13973403b4cef35bcc0fc4bb8f853bfcce26d0c578dd7cb6d52R835-R842

...but not the null case.

In the related project where I implemented useEmulator the same exact style is executed and this is exercised https://codecov.io/gh/invertase/react-native-firebase/src/4c688413cb6516ecfdbd4ea325103d0d8d8d84a8/packages/auth/lib/index.js so the style should work

As a follow on / with time available I was hoping to port the [necessary e2e auth-emulator-manipulation library](https://github.com/invertase/react-native-firebase/pull/4552/files#diff-44ccd5fb03b0d9e447820032866f2494c5a400a52873f0f65518d06aedafe302) from JS to Dart so the e2e auth test suite here can use the emulator in testing while still getting OOB codes / using disabled users etc but I was hoping to get the feature out first